packages/@aws-containers/aws-ecs-builder/lib/extensions/appmesh.ts

```diff
@@ -0,0 +1,326 @@
+          // the FireLens extension, so that outbound application logs don't have to go through Envoy
+          // and therefore add extra burden to the proxy sidecar. Instead the logs can go directly
+          // to CloudWatch
+          ignoredGID: 1338,
+
+          egressIgnoredIPs: [
+            '169.254.170.2', // Allow services to talk directly to ECS metadata endpoints
+            '169.254.169.254', // and EC2 instance endpoint
+          ],
+
+          // If there is outbound traffic to specific ports that you want to
+          // ignore the proxy those ports can be added here.
+          egressIgnoredPorts: [],
+        },
+      }),
+    } as ecs.TaskDefinitionProps;
+  }
+
+  public useTaskDefinition(taskDefinition: ecs.TaskDefinition) {
+    var region = cdk.Stack.of(this.scope).region;
+    var appMeshRepo;
+    var ownerAccount;
+
+    // This is currently necessary because App Mesh has different images in each region,
+    // and some regions have their images in a different account. See:
+    // https://docs.aws.amazon.com/app-mesh/latest/userguide/envoy.html
+    if (region === 'me-south-1') {
+      ownerAccount = APPMESH_ECR_ACCOUNTS.ME_SOUTH_1;
+    } else if (region === 'ap-east-1') {
+      ownerAccount = APPMESH_ECR_ACCOUNTS.AP_EAST_1;
+    } else {
+      ownerAccount = APPMESH_ECR_ACCOUNTS.DEFAULT;
+    }
+
+    appMeshRepo = ecr.Repository.fromRepositoryArn(
+      this.scope,
+      `${this.parentService.id}-envoy-repo`,
+      `arn:aws:ecr:us-east-1:${ownerAccount}:repository/aws-appmesh-envoy`,
+    );
+
+    this.container = taskDefinition.addContainer('envoy', {
+      image: ecs.ContainerImage.fromEcrRepository(appMeshRepo, APP_MESH_ENVOY_SIDECAR_VERSION),
+      essential: true,
+      environment: {
+        APPMESH_VIRTUAL_NODE_NAME: `mesh/${this.mesh.meshName}/virtualNode/${this.parentService.id}`,
+        AWS_REGION: cdk.Stack.of(this.parentService).region,
+        ENABLE_ENVOY_STATS_TAGS: '1',
+        ENABLE_ENVOY_DOG_STATSD: '1',
+      },
+      healthCheck: {
+        command: [
+          'CMD-SHELL',
+          'curl -s http://localhost:9901/server_info | grep state | grep -q LIVE',
+        ],
+        startPeriod: cdk.Duration.seconds(10),
+        interval: cdk.Duration.seconds(5),
+        timeout: cdk.Duration.seconds(2),
+      },
+      memoryReservationMiB: 128,
+      user: '1337',
+      logging: new ecs.AwsLogDriver({ streamPrefix: 'envoy' }),
+    });
+
+    // Modify the task definition role to allow the Envoy sidecar to get
+    // configuration from the Envoy control plane, for this particular
+    // mesh only.
+    const policy = new iam.Policy(this.scope, `${this.parentService.id}-envoy-to-appmesh`);
+
+    const statement = new iam.PolicyStatement();
+    statement.addResources(this.mesh.meshArn);
+    statement.addActions('appmesh:StreamAggregatedResources');
+
+    policy.addStatements(statement);
+    policy.attachToRole(taskDefinition.taskRole);
```

Comment:
I don't like this way of handling IAM, seems overly verbose and not super readable. How about:

```suggestion
    new iam.Policy(this.scope, `${this.parentService.id}-envoy-to-appmesh, {
      roles: [taskDefinition.taskRole],
      statements: [
        new iam.PolicyStatement({
          resources: [this.mesh.meshArn],
          actions: ['appmesh:StreamAggregatedResources'],
        }),
      ],
    });
```

Reply:
IAM policy creation adjusted as suggested